core/src/main/kotlin/gropius/model/architecture/ComponentVersion.kt

```diff
@@ -32,73 +36,41 @@ class ComponentVersion(
 ) : RelationPartner(name, description), Versioned, MutableTemplatedNode {
 
     companion object {
-        const val VISIBLE_SELF_DEFINED = "VISIBLE_SELF_DEFINED"
-        const val INVISIBLE_SELF_DEFINED = "INVISIBLE_SELF_DEFINED"
-        const val VISIBLE_DERIVED = "VISIBLE_DERIVED"
-        const val INVISIBLE_DERIVED = "INVISIBLE_DERIVED"
+        const val INTRA_COMPONENT_DEPENDENCY_SPECIFICATION = "INTRA_COMPONENT_DEPENDENCY_SPECIFICATION"
     }
 
     @NodeRelationship(BaseTemplate.USED_IN, Direction.INCOMING)
     @GraphQLDescription("The Template of this ComponentVersion")
     @FilterProperty
     @delegate:Transient
-    val template by NodeProperty<ComponentVersionTemplate>()
+    override val template by NodeProperty<ComponentVersionTemplate>()
 
     @NodeRelationship(Component.VERSION, Direction.INCOMING)
     @GraphQLDescription("The Component which defines this ComponentVersions")
     @FilterProperty
     @delegate:Transient
     val component by NodeProperty<Component>()
 
-    @NodeRelationship(Interface.COMPONENT, Direction.INCOMING)
-    @GraphQLDescription("Interfaces created by visible InterfaceSpecifications, can be used in Relations.")
-    @FilterProperty
-    @delegate:Transient
-    val interfaces by NodeSetProperty<Interface>()
-
     @NodeRelationship(Project.COMPONENT, Direction.INCOMING)
     @GraphQLDescription("Projects which include this ComponentVersion")
     @FilterProperty
     @delegate:Transient
     val includingProjects by NodeSetProperty<Project>()
 
-    @NodeRelationship(VISIBLE_SELF_DEFINED, Direction.OUTGOING)
-    @GraphQLDescription(
-        """InterfaceSpecifications which are defined by this Component(Version), and result in
-        visible Interfaces on this ComponentVersion
-        """
-    )
+    @NodeRelationship(InterfaceDefinition.COMPONENT_VERSION, Direction.INCOMING)
+    @GraphQLDescription("InterfaceSpecificationVersions on this ComponentVersion.")
     @FilterProperty
     @delegate:Transient
-    val visibleSelfDefinedInterfaceSpecificationVersions by NodeSetProperty<InterfaceSpecificationVersion>()
+    val interfaceDefinitions by NodeSetProperty<InterfaceDefinition>()
```

Comment:
Doc and code don't quite fit together

Reply:
fixed with b03e4b760ec1a47aa2ae2e8ba9eb3c00e93c2239